src/main/java/agh/ics/oop/Animal.java

```diff
@@ -0,0 +1,77 @@
+package agh.ics.oop;
+
+public class Animal {
+
+
+    private MapDirection Orientation = MapDirection.NORTH;
+    private Vector2d Position = new Vector2d(2,2);
+
+
+    @Override
+    public String toString() {
+        return "Orientation: " + this.Orientation + " Position: " + this.Position;
+    }
+
+    public Boolean isAt(Vector2d position) {
+        return this.Position.equals(position);
+
+    }
+
+    public MapDirection getOrientation() {
+        return Orientation;
+    }
+
+    public void move(MoveDirection direction)
+    {
+        //zm
+        if (direction == MoveDirection.RIGHT)
+            this.Orientation = this.Orientation.next();
+        else if (direction == MoveDirection.LEFT)
```

Comment:
lepiej mimo wszystko switchem, ale jeśli już if to polecam zawsze dodawać `{ }`, nawet przy jednolinijkowcach. To się lubi szybko mścić, gdy dochodzi kolejna linijka... ;p